modules/op-running-unpriveleged-builds-in-a-container.adoc

```diff
@@ -0,0 +1,62 @@
+:_content-type: PROCEDURE
+
+[id='op-running-unpriveleged-builds-in-a-container._{context}']
+= Running unprivileged builds in a container
+
+You can run builds as a non-root user inside a container to restrict specific container activities and enhance security.
+
+[INFO]
+====
+To run as a non-root user inside the container, your container must first add the `subuid` and `subgid` with the latest shadow-utils.
+====
+
+.Procedure
+
+. Use the public https://catalog.redhat.com/software/containers/ubi8/buildah/602686f7b16b1eb2e30807ee?container-tabs=dockerfile[Containerfile] image based on Red Hat UBI.
+
+. Force run the container as the “build” user to prevent it from being “randomed” which will not have the setup needed to run as non-root.
+
+. Add `securityContext`` to each build and push steps to directly edit or export to a specific namespace the buildah ClusterTask.
++
+[source,yaml,subs="attributes+"]
+----
+securityContext:
+    runAsUser: 1000
+----
+
+. Change the `mountPath` for the build to your local user container directory to use the image cache.
++
+[source,yaml,subs="attributes+"]
+----
+volumeMounts:
+    - name: varlibcontainers
+      mountPath: /home/build/.local/share/containers
+----
+
+. Test it using https://github.com/tektoncd/pipeline/blob/main/docs/workspaces.md#configmap[ConfigMap workspace] along with a sample DockerFile and a sample taskrun.
++
+[source,terminal,subs="attributes+"]
+----
+$ tkn tr logs -Lf buildah-run
+----
++
+The following output is displayed:
++
+[source,terminal]
+----
+[build] ++ id
+[build] Running as USER ID uid=1000(build) gid=1000(build) groups=1000(build),1000690000
+[build] + echo 'Running as USER ID uid=1000(build) gid=1000(build) groups=1000(build),1000690000'
+[build] + buildah --storage-driver=vfs bud --format=oci --tls-verify=true --no-cache -f ./Dockerfile -t image-registry.openshift-image-registry.svc:5000/test/buildahuser .
+[build] STEP 1: FROM registry.access.redhat.com/ubi8/ubi AS buildah-runner
+[build] Getting image source signatures
+[build] Checking if image destination supports signatures
+[build] Copying blob sha256:adffa69631469a649556cee5b8456f184928818064aac82106bd08bd62e51d4e
+[build] Copying blob sha256:26f1167feaf74177f9054bf26ac8775a4b188f25914e23bda9574ef2a759cce4
+[build] Copying config sha256:fca12da1dc30ed8e7d03afb84b287fc695673fff9c04bfcb2ff404b558670a36
+[build] Writing manifest to image destination
+[build] Storing signatures
+[build] STEP 2: RUN dnf -y update &&     dnf -y install git &&     dnf clean all
+[build] Updating Subscription Management repositories.
+[...]
+----
```

Comment:
Is there something specific the user should be looking for in this output?

Reply:
Nothing specific — It should just serve the user to understand better on what could be the expected output.

Reply:
@HarshCasper you can mark this as resolved.